From 004296e0e36dbe3919ca43129ee576fe20cc5f4f Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sun, 4 Aug 2013 18:23:59 +0200
Subject: [PATCH 36/65] acerhk: only allow numbers as parameter for the various
 procfs modifiers

* /proc/driver/acerhk/wirelessled = [0-1]
* /proc/driver/acerhk/blueled     = [0-1]
* /proc/driver/acerhk/led         = [0-10] (values >1 set blinking pattern)

"on" or the localised "an" are no longer recognised.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 30 +++++++++++++-----------------
 1 file changed, 13 insertions(+), 17 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -1231,21 +1231,19 @@ static int acerhk_proc_init(void)
 static int acerhk_proc_led(struct file *file, const char *buffer,
 			   unsigned long count, void *data)
 {
-	char str[4];
+	char str[2];
 	int len;
 
-	if (count > 4)
-		len = 4;
+	if (count > 2)
+		len = 2;
 	else
 		len = count;
 
 	if (copy_from_user(str, buffer, len))
 		return -EFAULT;
 
-	str[3] = '\0';
-	if ((
-	     (len >= 2) && (!strncmp(str, "on", 2) || !strncmp(str, "an", 2))
-	    ) || str[0] == '1')
+	str[1] = '\0';
+	if (str[0] == '1')
 		set_mail_led(1);
 	else
 		set_mail_led(0);
@@ -1256,20 +1254,19 @@ static int acerhk_proc_led(struct file *
 static int acerhk_proc_wirelessled(struct file *file, const char *buffer,
 				   unsigned long count, void *data)
 {
-	char str[4];
+	char str[2];
 	int len;
 
-	if (count > 4)
-		len = 4;
+	if (count > 2)
+		len = 2;
 	else
 		len = count;
 
 	if (copy_from_user(str, buffer, len))
 		return -EFAULT;
 
-	str[3] = '\0';
-	if (((len >= 2) && (!strncmp(str, "on", 2) || !strncmp(str, "an", 2)))
-	    || str[0] == '1') {
+	str[1] = '\0';
+	if (str[0] == '1') {
 		if (acerhk_model_features & TM_F_WLAN_EC1)
 			enable_wlan_ec_1();
 		else if (acerhk_model_features & TM_F_WLAN_EC2)
@@ -1290,8 +1287,8 @@ static int acerhk_proc_wirelessled(struc
 
 /* Modified by Antonio Cuni: added support for blinking
    possible values:
-   - off, 0:       led always off
-   - on, an,  1:   led alway on
+   - 0: led always off
+   - 1: led alway on
    - n (a number): led blinking; n is the delay between
    two changes of state, in jiffies; n must
    be > 50, to prevent the user from overloading
@@ -1330,8 +1327,7 @@ static int acerhk_proc_blueled(struct fi
 
 			start_blinking();
 		}
-	} else if (acerhk_blueled_blinking_delay == 1 || !strncmp(str, "on", 2)
-		   || !strncmp(str, "an", 2)) {
+	} else if (acerhk_blueled_blinking_delay == 1) {
 		stop_blinking();
 
 		if (acerhk_model_features & TM_F_BLUE_EC1)
